### PR TITLE
feat(database): support switching column type

### PR DESCRIPTION
**Pull request by zqran**
https://user-images.githubusercontent.com/15389209/228466586-b7de1511-acd8-42c1-991c-aabe931cd600.mov

### CODE REVIEW

**Comment by Flrande** [comment]
When switching column types, the table header does not update correspondingly. Is this part still under development?

**Comment by zqran** [comment]
> When switching column types, the table header does not update correspondingly. Is this part still under development?

Do you mean **column title**? The **column title** should't update when switch column type.

**Review by Flrande — APPROVED** [review]

File: packages/blocks/src/database-block/components/cell-container.ts [hunk 1]
```diff
@@ -30,9 +30,14 @@ export class DatabaseCellContainer
...
 
+  private _timerId: number | undefined = undefined;
+
   setValue(value: unknown) {
     if (value !== undefined) {
-      setTimeout(() => {
+      if (this._timerId !== undefined) {
+        window.clearTimeout(this._timerId);
+      }
+      this._timerId = window.setTimeout(() => {
```
Comment (doodlewind) [inline packages/blocks/src/database-block/components/cell-container.ts:40]: Why is this timeout here? Is this avoiding some anti-pattern?

**Review by doodlewind — CHANGES_REQUESTED** [review]
Fast progress indeed! But I still want to figure out the best practice on a few issues before landing this. I'll see what I can do.

**Comment by zqran** [comment]
> Fast progress indeed! But I still want to figure out the best practice on a few issues before landing this. I'll see what I can do.

Full support, I will refactor these after discussing and clarifying best practices.

**Review by doodlewind — APPROVED** [review]

File: packages/store/src/workspace/page.ts [hunk 2]
```diff
@@ -270,6 +279,42 @@ export class Page extends Space<FlatBlockMap> {
     });
   }
 
+  // TODO: remove hard coded types
+  updateColumnValue(columnId: string, newType: 'select' | 'rich-text') {
```
Comment (doodlewind) [inline packages/store/src/workspace/page.ts:283]: @zqran this is a generic workaround, we will further improve this

Comment (zqran) [reply packages/store/src/workspace/page.ts:283]: 👌

**Pull request merged**